Doc/c-api/typeobj.rst

```diff
@@ -654,8 +654,8 @@ and :c:type:`PyType_Type` effectively act as defaults.)
    the instance is still in existence, but there are no references to it.  The
    destructor function should free all references which the instance owns, free all
    memory buffers owned by the instance (using the freeing function corresponding
-   to the allocation function used to allocate the buffer), and finally (as its
-   last action) call the type's :c:member:`~PyTypeObject.tp_free` function.  If the type is not
+   to the allocation function used to allocate the buffer), and then call the
+   type's :c:member:`~PyTypeObject.tp_free` function.  If the type is not
```

Comment:
Remove the `then`. Perhaps even include a short blurb saying that the instance might be resurrected by `tp_del` or `tp_finalize`.

Reply:
Removed the `then`, I really don't know enough about type resurrection to feel confident talking about it so I'm gonna punt that to someone more capable (maybe yourself :))